app/adapters/contributor.ts

```diff
@@ -12,42 +9,47 @@ export default class ContributorAdapter extends OsfAdapter {
     buildURL(
         modelName: 'contributor',
         id: string,
-        snapshot: DS.Snapshot,
+        snapshot: DS.Snapshot<'contributor'>,
         requestType: string,
     ) {
-        if (requestType === 'createRecord' || requestType === 'findRecord') {
-            let userId;
-            let nodeId;
-
-            if (requestType === 'findRecord') {
-                [nodeId, userId] = (id || '').split('-');
-                assert(`"contributorId" must be "nodeId-userId": got ${nodeId}-${userId}`, Boolean(nodeId && userId));
+        if (requestType === 'findRecord') {
+            const [objectId, userId] = (id || '').split('-');
+            let baseUrl;
+            assert(`"contributorId" must be "objectId-userId": got ${objectId}-${userId}`, Boolean(objectId && userId));
+            if (objectId.length === 5) {
```

Comment:
True. I don't think this can be resolved until we have a top level endpoint for querying contributors. Otherwise, we have to know whether it's a `node` or `draft_registration` instance the contributor is attached before building the url to fetch that `contributor` instance.

Reply:
It is always possible that either the node or the draftRegistration is not yet loaded when we try to find a contributor, albeit not in this case. Generally speaking, querying for an object should not rely on the object's related model to be loaded. The solution should be implementing a v2/contributors endpoint for CRUD ops of contributors while this stays temporary.

Reply:
> querying for an object should not rely on the object's related model to be loaded.
This is warranted in this case because it's a polymorphic relationship we don't want to treat as one for some reasons. As of now, I don't know of any plans from platform to create first-class endpoint for contributors.

Reply:
We should ask PLAT to create a top-level endpoint for contributors.

Reply:
@adlius A better way to handle this for now would be to specify an `objectType` in the findRecord call as such  `store.findRecord('contributor', contributorId, { adapterOptions: { objectType }})` where `objectType` can be `draft-registration` or `node`. That way we can use `snapshot.adapterOptions.objectType` and not have to guess based on lengths or peek the store twice. Same can be done for `createRecord`. We need only delete `snapshot.adapterOptions.objectType` that way it's not appended to the url.

Reply:
I didn't notice that we're already peeking the store for node/draft. In which case peeking the store twice should be fine.